Text/HealthcareTA/HealthcareTA.cs

```diff
@@ -0,0 +1,136 @@
+                            Message = $"Healthcare Text Analytics Error: The submitted document was over {maxCharLength} characters. It has been truncated to fit this requirement."
+                        });
+                        document = document.Substring(0, maxCharLength);
+                    }
+                    List<string> batchInput = new List<string>()
+                    {
+                        document
+                    };
+
+                    // start analysis process
+                    var timer = System.Diagnostics.Stopwatch.StartNew();
+                    AnalyzeHealthcareEntitiesOperation healthOperation = await client.StartAnalyzeHealthcareEntitiesAsync(batchInput, "en", options);
+                    var healthOperationTask = healthOperation.WaitForCompletionAsync().AsTask();
+
+                    if (await Task.WhenAny(healthOperationTask, Task.Delay(timeoutMiliseconds)) == healthOperationTask)
+                    {
+                        // Task Completed, now lets process the result.
+                        outRecord.Data["status"] = healthOperation.Status.ToString();
+                        if (healthOperation.Status != TextAnalyticsOperationStatus.Succeeded)
+                        {
+                            // The operation was not a success
+                            outRecord.Warnings.Add(new WebApiErrorWarningContract
+                            {
+                                Message = "Healthcare Text Analytics Error: Health Operation returned a non-succeeded status."
+                            });
+                        }
+                        else
+                        {
+                            // The operation was a success, so lets add the results to our output.
+                            await ExtractEntityData(healthOperation.Value, outRecord);
+                        }
+                    }
+                    else
+                    {
+                        // Timeout
+                        outRecord.Warnings.Add(new WebApiErrorWarningContract
+                        {
+                            Message = "Healthcare Text Analytics Error: The Text Analysis Operation took too long to complete."
+                        });
+                    }
+
+                    // Record how long this task took to complete.
+                    timer.Stop();
+                    outRecord.Data["timeToComplete"] = timer.Elapsed.TotalSeconds.ToString();
+
+                    return outRecord;
+                });
+
+            return new OkObjectResult(response);
+        }
+
+        private static async Task ExtractEntityData(AsyncPageable<AnalyzeHealthcareEntitiesResultCollection> pages, WebApiResponseRecord outRecord)
+        {
+            await foreach (AnalyzeHealthcareEntitiesResultCollection documentsInPage in pages)
+            {
+                foreach (AnalyzeHealthcareEntitiesResult entitiesInDoc in documentsInPage)
```

Comment:
>AnalyzeHealthcareEntitiesResult [](http://example.com/codeflow?start=25&length=31)

This does have a Warnings field so we should add those to the output as well.

Reply:
done and done 👍